src/rate_limiter/src/lib.rs

```diff
@@ -549,26 +602,28 @@ pub(crate) mod tests {
         let refill_ms = 1000;
         let mut tb = TokenBucket::new(capacity, 0, refill_ms as u64).unwrap();
 
-        assert!(tb.reduce(123));
+        assert!(tb.reduce(123) == BucketReduction::Success);
```

Comment:
consistency nit:
```suggestion
        assert_eq!(tb.reduce(123), BucketReduction::Success);
```

Same for all below.